library/ssl_cli.c

```diff
@@ -974,116 +1237,82 @@ static int ssl_write_client_hello( mbedtls_ssl_context *ssl )
         MBEDTLS_SSL_DEBUG_MSG( 3, ( "client hello, compress alg.: %d %d",
                             MBEDTLS_SSL_COMPRESS_DEFLATE, MBEDTLS_SSL_COMPRESS_NULL ) );
 
-        *p++ = 2;
-        *p++ = MBEDTLS_SSL_COMPRESS_DEFLATE;
-        *p++ = MBEDTLS_SSL_COMPRESS_NULL;
+        if( buflen <   1 /* Length byte                 */
```

Comment:
This could be extracted either to a define or a variable. If the buf += 3 below is related to this - it could use the same variable.

Reply:
Ok, changed it, although I don't think it adds much in this simple case.